torch/csrc/distributed/autograd/utils.h

```diff
@@ -34,19 +34,23 @@ TORCH_API DistAutogradContext* addRecvRpcBackward(
 
 // This method is a wrapper utility used internally to wrap autograd info
 // and attach autograd function for each type of rpc call if it has valid
-// context and tensors require grads, in this case, return RpcWithAutograd
-// message; otherwise return original rpc message.
+// context and tensors require grads or checkRequiresGrad is false, in this
+// case, return RpcWithAutograd message; otherwise return original rpc message.
+// NB: checkRequiresGrad is useful when the request does not contain any tensor
+// but the corresponding response does.
 TORCH_API rpc::Message getMessageWithAutograd(
     const rpc::worker_id_t dstId,
     rpc::Message&& wrappedRpcMsg,
-    rpc::MessageType msgType);
+    rpc::MessageType msgType,
+    bool checkRequiresGrad = true);
```

Comment:
nit: maybe call this `forceGradRecording` and default it false. This way `ScriptRRefFetchCall` can set this to true, to force recording grad even if tensors are empty.